modules/core/src/test/java/org/apache/ignite/cache/IgniteCacheEntryProcessorSequentialCallTest.java

```diff
@@ -0,0 +1,108 @@
+package org.apache.ignite.cache;
+
+import org.apache.ignite.Ignite;
+import org.apache.ignite.IgniteCache;
+import org.apache.ignite.configuration.CacheConfiguration;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.apache.ignite.transactions.Transaction;
+import org.apache.ignite.transactions.TransactionConcurrency;
+import org.apache.ignite.transactions.TransactionIsolation;
+import javax.cache.processor.EntryProcessorException;
+import javax.cache.processor.MutableEntry;
+
+/**
+ */
+public class IgniteCacheEntryProcessorSequentialCallTest extends GridCommonAbstractTest {
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        return super.getConfiguration(igniteInstanceName)
+            .setCacheConfiguration(new CacheConfiguration("cache")
+                .setCacheMode(CacheMode.PARTITIONED)
+                .setAtomicityMode(CacheAtomicityMode.TRANSACTIONAL)
+                .setRebalanceMode(CacheRebalanceMode.SYNC)
+                .setWriteSynchronizationMode(CacheWriteSynchronizationMode.FULL_SYNC)
+                .setMaxConcurrentAsyncOperations(0)
+                .setBackups(0));
+    }
+
+    /**
+     * Test for sequential entry processor invoking not null value.
+     */
+    public void testSequentialCallOnOneNode() throws Exception {
+        Ignite ignite = startGrid(0);
+        try {
+            IgniteCache<TestKey, TestValue> cache = ignite.cache("cache");
+
+            TestKey key = new TestKey(1L);
+            TestValue val = new TestValue();
+            val.value("1");
+
+            cache.put(key, val);
+
+            NotNullCacheEntryProcessor cacheEntryProcessor = new NotNullCacheEntryProcessor();
+
+            try (Transaction transaction = ignite.transactions().txStart(TransactionConcurrency.OPTIMISTIC,
+                TransactionIsolation.SERIALIZABLE, 20000L, 1)) {
+
+                cache.invoke(key, cacheEntryProcessor);
+                cache.invoke(key, cacheEntryProcessor);
+
+                transaction.commit();
+            }
+        }
+        finally {
```

Comment:
I think that better to use try-with-resources
try (Ignite node = startGrid(0)) {
without explicit finally statement.

Reply:
fixed